README.md

```diff
@@ -93,52 +129,51 @@ This RFC makes no requirement as to how the transition should happen, but here a
-  - Slower evaluation, since a lot more directories need to be traversed
-  - Increased end-user complexity:
-    - Creating the package files often requires the creating of 2 directories, not just one
-    - Referencing the files requires knowing the prefix schema
-- Improve deprecation signalling by creating `.nix` files that act like a symlink, but with a warning. Something like this:
-  ```nix
-  builtins.trace "warning: Using deprecated path ${./.}, use pkgs/unit/<name> instead, this will be removed after NixOS 22.05"
-    (import ../../pkgs/unit/name)
-  ```
-  The main downside of this is the increased complexity of implementation
-
- - Use `package.nix` instead of `pkg-fun.nix`
-   - Makes the migration to a non-function form of overridable packages harder in the future. We'd like to use `package.nix` for a package format that's based on a fixpoint rather than a function, because that will make overriding simpler.
-
- - Use `default.nix` instead of `pkg-fun.nix`
-   - `default.nix`'s only benefits do not apply
-     - removing the need to specify the file name in expressions, but this does not apply because we have to do this at most once in the code that replaces definitions from `all-packages.nix`.
-     - removing the need to specify the file name on the command line, but this does not apply because a package function must be imported into an expression before it can be used, making `nix build -f pkgs/unit/hello` equally broken regardless of file name.
-   - Choosing `default.nix` would bias the purpose of the `pkg` directory to serve only as package definitions, whereas we could make the tree more human friendly by grouping files together by "topic" rather than by technical delineations. For instance, having a package definition, changelog, package-specific config generator and perhaps even NixOS module in one directory makes work on the package in a broad sense easier. This is not a goal of this RFC, but a motivation to make this a future possibility.
-   - `pkg-fun.nix` frees up `default.nix` for a short expression that is actually buildable, e.g. `(import ../..).hello`.
-
- - Use `unit/` instead of `pkgs/unit`. This is future proof in case we want to
-   make the directory structure more general purpose, but this is out of scope
-   and we want to improve tooling to make renames easy.
-
+- Use a flat directory `pkgs/unit/*/pkg-fun.nix` instead, arguments:
+  - Good because it's simpler, both for the user and for the code
+  - Good because it speeds up Nix evaluation since there's only a single directory to call `builtins.readDir` on instead of many
+    - With an optimized `readDir` this isn't much of a problem
+  - Bad because it causes GitHub to limit the rendering of that directory to 1'000 entries (and we have about 10'000 that benefit from this transition for a start)
```

Comment:
```suggestion
  - Bad because the GitHub web interface only renders the first 1'000 entries (and we have about 10'000 that benefit from this transition, even given the restrictions)
```